File: codeforces/63.spy_detected!/spy_detected!.py
# Author: José Rodolfo (jric2002)
t = int(input())
while (t):
  n = int(input())
  a = list(map(int, input().split(" ")))
  index = 0
  counter = 1
  number = a[0]
  for i in range(1, n):
    if ((number != a[i]) and (index < 1)):
      index = i + 1
    if (number == a[i]):
      counter += 1
  if (counter == 1):
    index = index - counter
  print(index)
  t -= 1

File: codeforces/low_difficulty/difficulty_800/63.spy_detected!_[1512A]/spy_detected!.cpp
#include <iostream>
/* Author: José Rodolfo (jric2002) */
using namespace std;
int main() {
  unsigned short int t, n, a, index, number, counter;
  cin >> t;
  while (t--) {
    cin >> n;
    index = 0;
    counter = 1;
    cin >> number;
    for (unsigned short int i = 2; i <= n; i++) {
      cin >> a;
      if (number != a && index < 1) {
        index = i;
      }
      if (number == a) {
        counter += 1;
      }
    }
    if (counter == 1) {
      index = index - counter;
    }
    cout << index << endl;
  }
  return 0;
}